scripts/generateReleasePr.mjs

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env zx
+
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import { $, chalk } from 'zx';
+
+// Exit when error
+$.verbose = false;
+
+const args = process.argv.slice(2);
+const bumpTypeArgs = args.find((arg) => arg.startsWith('--type='));
+
+async function getCurrentVersion() {
+  const packageJsonPath = path.join(
+    process.cwd(),
+    'packages/core/package.json',
+  );
+  const packageJson = JSON.parse(await fs.readFile(packageJsonPath, 'utf-8'));
+  return packageJson.version;
+}
+
+async function getNextVersion(currentVersion, type) {
+  const [major, minor, patch] = currentVersion.split('.').map(Number);
+  switch (type) {
+    case 'patch':
+      return `${major}.${minor}.${patch + 1}`;
+    case 'minor':
+      return `${major}.${minor + 1}.0`;
+    case 'major':
+      return `${major + 1}.0.0`;
+    default:
+      throw new Error('Invalid version type');
+  }
+}
+
+async function generateChangesetFile(bumpType, nextVersion) {
+  const changesetDir = path.join(process.cwd(), '.changeset');
+  const timestamp = Date.now();
+  const filename = `${timestamp}-${bumpType}-release.md`;
+  const content = `---
+"@rslib/core": ${bumpType}
+---
+
+Release version ${nextVersion}
+`;
+
+  await fs.mkdir(changesetDir, { recursive: true });
+  await fs.writeFile(path.join(changesetDir, filename), content);
+
+  console.log(chalk.blue(`Generated changeset file: ${filename}`));
+}
+
+async function main() {
+  try {
+    // 1. Read the current version
+    const currentVersion = await getCurrentVersion();
+    console.log(chalk.blue(`Current version: ${currentVersion}`));
+
+    // 2. Determine bump type
+    const bumpType = bumpTypeArgs ? bumpTypeArgs.split('=')[1] : 'patch';
```

Comment:
We could use https://nodejs.org/api/util.html#utilparseargsconfig, supported since node 16.